gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageFileSystem.java

```diff
@@ -324,6 +340,17 @@ public SeekableByteChannel open(URI path, GoogleCloudStorageReadOptions readOpti
     return gcs.open(resourceId, readOptions);
   }
 
+  public SeekableByteChannel open(GoogleCloudStorageItemInfo itemInfo, GoogleCloudStorageReadOptions readOptions)
+          throws IOException {
+    logger.atFiner().log("open(path: %s, readOptions: %s)", itemInfo, readOptions);
+    Preconditions.checkNotNull(UriPaths.fromResourceId(itemInfo.getResourceId(),false), "path should not be null");
```

Comment:
Can we add another Precondition to check notNull for `itemInfo` ?

Reply:
Yes.